Fix Depwarns

SimonDanisch opened the pull request:
I just tried using WinRPM on 0.4, not sure if that's a sufficient test... Actually, the depwarns are so bad, that WinRPM just stops installing binaries without any error. So this also needs the depwarn removal from LibExpat (amitmurthy/LibExpat.jl#45)

timholy commented:
LGTM. OK to merge?

timholy commented:
It does seem rather important to test this on AppVeyor :smile:. I'll leave others to it.

tkelman commented:
Oh and Julia 0.3 is apparently still supported here too but that doesn't need to remain the case.

tkelman commented on src/WinRPM.jl:
```diff
@@ -349,7 +350,7 @@ end
 install(pkg::AbstractString, arch::AbstractString=OS_ARCH; yes = false) = install(select(lookup(pkg, arch),pkg); yes = yes)
 
-function install(pkgs::Vector{ASCIIString}, arch = OS_ARCH; yes = false)
+function install(pkgs::Vector{String}, arch = OS_ARCH; yes = false)
```
I don't think this will work on 0.4 or earlier, `Compat.ASCIIString` is less risky

SimonDanisch commented:
Okay, removed support for 0.3 and squashed...
Tested on 0.4 and 0.5 with:
```
using WinRPM
WinRPM.update()
WinRPM.install("gcc")
```
Would it make sense to add this as a minimal test for appveyor and travis?

tkelman commented on src/WinRPM.jl:
```diff
@@ -1,17 +1,23 @@
+import Compat.String
+
+if is_unix()
+  using HTTPClient.HTTPC
```
4 space indent

tkelman commented:
yeah something simple would be better than nothing. maybe a few bindeps integration tests via `test/REQUIRE` would also be good

tkelman commented:
needs rebase

SimonDanisch commented:
Allrighty! Hope this is ready now :D

tkelman commented on src/WinRPM.jl:
```diff
@@ -42,8 +48,13 @@ function __init__()
-@unix_only download(source::AbstractString) = (x=HTTPC.get(source); (bytestring(x.body),x.http_code))
-@windows_only function download(source::AbstractString; retry = 5)
+if is_unix()
+  function download(source::AbstractString)
```
4-space please

tkelman commented on src/WinRPM.jl:
```diff
@@ -1,18 +1,24 @@
 
-if OS_NAME == :Windows
-    const OS_ARCH = WORD_SIZE == 64 ? "mingw64" : "mingw32"
+if is_windows()
```
minimum Compat version probably needs increasing

SimonDanisch replied:
`is_windows` seems to be in 0.7.16, from what I can tell!

tkelman replied:
0.7.20 https://github.com/JuliaLang/Compat.jl/commit/85f886227c332507ec5b31627e0b3a9fbb6aa274

tkelman commented on src/WinRPM.jl:
```diff
@@ -455,15 +469,17 @@ function do_install(package::Package)
-                end
+            if is_unix()
+                cd(installdir) do
+                  if success(`rpm2cpio $path2` | `cpio -imud`)
```
here as well

SimonDanisch replied:
ugh, my editor settings are messed up on windows -.-

tkelman replied:
if your editor is Juno, the mismatch of default indent there relative to the standard julia style is a continued bug that users of that editor should continue reporting to its author

tkelman commented on src/WinRPM.jl:
```diff
@@ -42,25 +48,33 @@ function __init__()
     end
-    return "",0
+else
+    error("Platform not supported: $(Sys.KERNEL)")
```
`Compat.KERNEL`, otherwise won't work on 0.4 (not that this line is likely)

tkelman commented on src/WinRPM.jl:
```diff
@@ -455,15 +469,17 @@ function do_install(package::Package)
+                    if success(`rpm2cpio $path2` | `cpio -imud`)
+                        err = nothing
+                    end
+              end
```
if you want to give me commit access to your fork I can also fix these

tkelman commented:
Thanks for bearing with my pickiness. I'm indirectly testing this at https://ci.appveyor.com/project/tkelman/lightxml-jl/build/1.0.22

SimonDanisch commented:
Thanks Tony! Sorry for these messy fixes inbetween trains ;)

tkelman commented:
I'll take whatever help I can get with this stuff!

tkelman commented:
There's still a method overwrite warning, but looks like this works.
